researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java

```diff
@@ -386,7 +386,8 @@ public Completable signIn(@NonNull String email, @NonNull String password) {
     }
 
     public boolean isSignedIn() {
-        return authenticationManager.getEmail() != null;
+        return authenticationManager.getEmail() != null &&
+                authenticationManager.isConsentedOnBridgeServer();
```

Comment:
This fixes AA-125 by adding a check for consent on the bridge (through the web service returned UserSession) that if not hit, will push the user to the email verification screen

Reply:
IIRC from iOS discussions, the way to detect email verification is the correct next step, is to know the user has signed up, yet the email/password used to sign in failed. 

I know there's been updates to Auth, so I'll look into whether we now provide a real API to know sign in failed due to lack of email verification, and not incorrect email/password

Reply:
I thought the call to sign up created a UserSession, but it doesn't look like it does.  You are correct then, that would be a better test for if email is verified.